test/functional/rpc_getdescriptoractivity.py

```diff
@@ -0,0 +1,164 @@
+from test_framework.wallet import MiniWallet, getnewdestination
+
+
+class GetBlocksActivityTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+
+    def run_test(self):
+        node = self.nodes[0]
+        wallet = MiniWallet(node)
+        node.setmocktime(node.getblockheader(node.getbestblockhash())['time'])
+        wallet.generate(200, invalid_call=False)
+
+        self.test_no_activity(node)
+        self.test_activity_in_block(node, wallet)
+        self.test_no_mempool_inclusion(node, wallet)
+        self.test_multiple_addresses(node, wallet)
+        self.test_invalid_blockhash(node, wallet)
+        self.test_confirmed_and_unconfirmed(node, wallet)
+        self.test_receive_then_spend(node, wallet)
+
+    def test_no_activity(self, node):
+        _, spk_1, addr_1 = getnewdestination()
+        result = node.getdescriptoractivity([], [f"addr({addr_1})"], True)
+        assert_equal(len(result['activity']), 0)
+
+    def test_activity_in_block(self, node, wallet):
+        _, spk_1, addr_1 = getnewdestination()
+        txid = wallet.send_to(from_node=node, scriptPubKey=spk_1, amount=1 * COIN)['txid']
+        blockhash = self.generate(node, 1)[0]
+
+        # Test getdescriptoractivity with the specific blockhash
+        result = node.getdescriptoractivity([blockhash], [f"addr({addr_1})"], True)
+
+        for k, v in {
+                'address': addr_1,
+                'amount': Decimal('1.00000000'),
+                'blockhash': blockhash,
+                'desc': 'rawtr',  # partial
+                'height': 201,
+                'scriptpubkey_hex': spk_1.hex(),
+                'txid': txid,
+                'type': 'receive',
+                'vout': 1,
+        }.items():
+            if k == 'desc':
+                assert_equal(result['activity'][0][k].split('(')[0], v)
+            else:
+                assert_equal(result['activity'][0][k], v)
+
+
+    def test_no_mempool_inclusion(self, node, wallet):
+        _, spk_1, addr_1 = getnewdestination()
+        wallet.send_to(from_node=node, scriptPubKey=spk_1, amount=1 * COIN)
+
+        _, spk_2, addr_2 = getnewdestination()
+        wallet.send_to(
+            from_node=node, scriptPubKey=spk_2, amount=1 * COIN)
+
+        # Do not generate a block to keep the transaction in the mempool
+
+        result = node.getdescriptoractivity([], [f"addr({addr_1})", f"addr({addr_2})"], False)
+
+        assert_equal(len(result['activity']), 0)
+
+    def test_multiple_addresses(self, node, wallet):
+        _, spk_1, addr_1 = getnewdestination()
+        _, spk_2, addr_2 = getnewdestination()
+        wallet.send_to(from_node=node, scriptPubKey=spk_1, amount=1 * COIN)
+        wallet.send_to(from_node=node, scriptPubKey=spk_2, amount=2 * COIN)
+
+        blockhash = self.generate(node, 1)[0]
+
+        result = node.getdescriptoractivity([blockhash], [f"addr({addr_1})", f"addr({addr_2})"], True)
```

Comment:
would be great if there was a case showing flipping descriptor ordering doesn't change result ordering

Reply:
another nice case would be demonstrating the order of activity results intra-block (order in block seems to be respected)

can either set fees differentially, or explicitly construct a block via `node.rpc.generateblock` to be quick about it

Reply:
Added testcases flipping both descriptor order and blockhash order to ensure results don't change.